src/overlays/actors/ovl_Door_Killer/z_door_killer.c

```diff
@@ -68,41 +84,438 @@ static ColliderJntSphInit D_80996000 = {
+               (fabsf(playerPosRelToDoor.x) < 20.0f) && (playerPosRelToDoor.z < 50.0f) &&
+               (playerPosRelToDoor.z > 0.0f)) {
+        // Set player properties to make the door openable if within range
+        angleToFacingPlayer = player->actor.shape.rot.y - this->actor.shape.rot.y;
+        if (playerPosRelToDoor.z > 0.0f) {
+            angleToFacingPlayer = 0x8000 - angleToFacingPlayer;
+        }
+        if (ABS(angleToFacingPlayer) < 0x3000) {
+            player->doorType = PLAYER_DOORTYPE_FAKE;
+            player->doorDirection = (playerPosRelToDoor.z >= 0.0f) ? 1.0f : -1.0f;
+            player->doorActor = &this->actor;
+        }
+    }
+    DoorKiller_SetAC(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Killer/func_80995368.s")
+/**
+ * Grabs the virtual address of the texture from the relevant door object
+ */
+void DoorKiller_UpdateTexture(Actor* thisx, GlobalContext* globalCtx) {
+    DoorKiller* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Killer/func_80995518.s")
+    gSegments[6] = VIRTUAL_TO_PHYSICAL(globalCtx->objectCtx.status[this->doorObjBankIndex].segment);
+    this->texture = SEGMENTED_TO_VIRTUAL(this->texture);
+    gSegments[6] = VIRTUAL_TO_PHYSICAL(globalCtx->objectCtx.status[thisx->objBankIndex].segment);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Killer/func_809958E4.s")
+/**
+ * Gets the correct door texture, defines the appropriate draw fucntion and action function based on type behaviour
+ * (door or rubble).
+ */
+void DoorKiller_SetProperties(DoorKiller* this, GlobalContext* globalCtx) {
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->doorObjBankIndex)) {
+        DoorKiller_UpdateTexture(&this->actor, globalCtx);
+        switch (this->actor.params & 0xFF) {
+            case DOOR_KILLER_DOOR:
+                this->actionFunc = DoorKiller_Wait;
+                this->actor.draw = DoorKiller_DrawDoor;
+                break;
+            case DOOR_KILLER_RUBBLE_PIECE_1:
+            case DOOR_KILLER_RUBBLE_PIECE_2:
+            case DOOR_KILLER_RUBBLE_PIECE_3:
+            case DOOR_KILLER_RUBBLE_PIECE_4:
+                this->actionFunc = DoorKiller_FallAsRubble;
+                this->actor.draw = DoorKiller_DrawRubble;
+                break;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Killer/func_80995A50.s")
+void DoorKiller_Update(Actor* thisx, GlobalContext* globalCtx) {
+    DoorKiller* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Killer/func_80995A84.s")
+    this->actionFunc(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Killer/func_80995CDC.s")
+void DoorKiller_SetTexture(Actor* thisx, GlobalContext* globalCtx) {
+    DoorKiller* this = THIS;
+    s32 doorTexture = this->texture;
```

Comment:
```suggestion
    u64* doorTexture = this->texture;
```
textures all have the type `u64*`